### PR TITLE
feat: add golang runtime metrics, process metrics and std lib db metrics

### PR DESCRIPTION
**Pull request by sysvm**
### Description

Add golang runtime metrics, process metrics and std lib db metrics to improve availability.

### Rationale

Improve availability and speed to debug problems.

### Example

`curl http://localhost:24367/metrics`

```
process metrics:
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 7.31
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 8192
......

golang metrics:
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 0
go_gc_duration_seconds{quantile="0.25"} 0
go_gc_duration_seconds{quantile="0.5"} 0
go_gc_duration_seconds{quantile="0.75"} 0
go_gc_duration_seconds{quantile="1"} 0
go_gc_duration_seconds_sum 0
go_gc_duration_seconds_count 0
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 7
......

db metrics:
# HELP go_sql_idle_connections The number of idle connections.
# TYPE go_sql_idle_connections gauge
go_sql_idle_connections{db_name="spdb"} 3
# HELP go_sql_in_use_connections The number of connections currently in use.
# TYPE go_sql_in_use_connections gauge
go_sql_in_use_connections{db_name="spdb"} 0
......
```

### Changes

Notable changes: 
* N/A

### CODE REVIEW

File: pkg/metrics/metric_items.go [hunk 1]
```diff
@@ -84,6 +87,11 @@ var MetricsItems = []prometheus.Collector{
 	MigrateGVGCounter,
 	MigrateObjectTimeHistogram,
 	MigrateObjectCounter,
+
+	// golang runtime and process metrics
+	collectors.NewGoCollector(collectors.WithGoCollectorRuntimeMetrics(collectors.GoRuntimeMetricsRule{
+		Matcher: regexp.MustCompile("/.*")})),
+	collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
```
Comment (will-2012) [inline pkg/metrics/metric_items.go:94]: Consistent with the above, define a variable.

**Review by will-2012 — APPROVED** [review]

File: pkg/metrics/metric_items.go (hunk 1 above)
Comment (sysvm) [reply pkg/metrics/metric_items.go:94]: These two metrics are derived from function.

File: base/gfspapp/app_options.go [hunk 2]
```diff
@@ -256,6 +256,13 @@ func DefaultGfSpDBOption(app *GfSpBaseApp, cfg *gfspconfig.GfSpConfig) error {
 				return
 			}
 			app.gfSpDB = db
```
[the remark this answers is not in the document]
Comment (sysvm) [reply base/gfspapp/app_options.go:258]: fixed

[the remark this answers is not in the document]
Comment (sysvm) [reply base/gfspapp/app_options.go]: db is not modified by RegisterStdDBStats, this function is just used to get some db metrics.

**Pull request merged**